packages/aws-cdk-lib/aws-kms/lib/key.ts

```diff
@@ -715,6 +720,17 @@ export class Key extends KeyBase {
         KeySpec.SYMMETRIC_DEFAULT,
         KeySpec.SM2,
       ],
+      [KeyUsage.KEY_AGREEMENT]: [
```

Comment:
I confirmed from the documentation and management console that only the following keySpecs are supported, and added the unsupported ones to the denyList.

> * For asymmetric KMS keys with NIST-recommended elliptic curve key pairs, specify SIGN_VERIFY or KEY_AGREEMENT.
>* For asymmetric KMS keys with SM2 key pairs (China Regions only), specify ENCRYPT_DECRYPT, SIGN_VERIFY, or KEY_AGREEMENT.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kms-key.html#cfn-kms-key-keyusage